frontend/src/app/modules/boards/board/add-list-modal/add-list-modal.component.ts

```diff
@@ -38,21 +38,56 @@ import { BoardActionsRegistryService } from "core-app/modules/boards/board/board
 import { BoardActionService } from "core-app/modules/boards/board/board-actions/board-action.service";
 import { HalResource } from "core-app/modules/hal/resources/hal-resource";
 import { AngularTrackingHelpers } from "core-components/angular/tracking-functions";
-import { CreateAutocompleterComponent } from "core-app/modules/autocompleter/create-autocompleter/create-autocompleter.component.ts";
+import { Observable, of } from "rxjs";
 import { DebouncedRequestSwitchmap, errorNotificationHandler } from "core-app/helpers/rxjs/debounced-input-switchmap";
 import { ValueOption } from "core-app/modules/fields/edit/field-types/select-edit-field/select-edit-field.component";
 import { HalResourceNotificationService } from "core-app/modules/hal/services/hal-resource-notification.service";
+import {OpAutocompleterComponent} from "core-app/modules/autocompleter/op-autocompleter/op-autocompleter.component";
+import { APIV3Service } from "core-app/modules/apiv3/api-v3.service";
+import { CurrentProjectService } from "core-components/projects/current-project.service";
 
 @Component({
   templateUrl: './add-list-modal.html'
 })
 export class AddListModalComponent extends OpModalComponent implements OnInit {
-  /** Keep a switchmap for search term and loading state */
-  public requests = new DebouncedRequestSwitchmap<string, ValueOption>(
-    (searchTerm:string) => this.actionService.loadAvailable(this.board, this.active, searchTerm),
-    errorNotificationHandler(this.halNotification),
-    true
-  );
+
+  @ViewChild(OpAutocompleterComponent, { static: true }) public ngSelectComponent:OpAutocompleterComponent;
+  
+  data = (searchTerm:string):Observable<HalResource[]> => {
+
+    // Remove prefix # from search
+
+    searchTerm = searchTerm.replace(/^#/, '');
+    let filteredData = this.actionService.loadAvailable(this.board, this.active, searchTerm);
+
```

Comment:
In order to perform side effects (actions that affect outside of the observable stream) you should use the `tap` operator. It would be also more clear if all the warning logic would be extracted to a method (e.g. warnIfNoOptions):
`return this.actionService.loadAvailable(this.board, this.active, searchTerm).pipe(tap(values => this.warnIfNoOptions(values)));`